src/arrayMethodJoin.js

```diff
@@ -6,6 +6,37 @@
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
     // write code here
+    let returnString = '';
+    let separatorCopy = separator;
+    const thisCopy = [...this];
+
+    // Check if our separator is undefined or null,
+    // so we change it to value we need
+
+    if (separator === undefined) {
+      separatorCopy = ',';
+    } else if (separator === null) {
+      separatorCopy = 'null';
+    }
+
+    for (let item = 0; item < this.length - 1; item++) {
+      // Change all ignored characters to separator(
+      // f.e undefined or empty string, but exclude false, null, NaN)
+      if (!this[item] && typeof this[item] !== 'number'
+      && this[item] !== false) {
+        thisCopy[item] = separatorCopy;
```

Comment:
I think u can do it in one line;
thisCopy[item] = separatorCopy;
returnString += thisCopy[item];
-> returnString += separatorCopy;